build-scripts/xc-universal-binary.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+set -euvx
+
+# This should be invoked from inside xcode, not manually
+if [ "$#" -ne 4 ]
+then
+    echo "Usage (note: only call inside xcode!):"
+    echo "path/to/build-scripts/xc-universal-binary.sh <STATIC_LIB_NAME> <FFI_TARGET> <APPSVC_ROOT_PATH> <buildvariant>"
+    exit 1
+fi
+# e.g. liblogins_ffi.a
+STATIC_LIB_NAME=$1
+# what to pass to cargo build -p, e.g. logins_ffi
+FFI_TARGET=$2
+# path to app services root
+APPSVC_ROOT=$3
+# buildvariant from our xcconfigs
+BUILDVARIANT=$4
+
+RELFLAG=
+RELDIR="debug"
+if [[ "$BUILDVARIANT" != "debug" ]]; then
+    RELFLAG=--release
+    RELDIR=release
+fi
+
+LIBSDIR=$APPSVC_ROOT/libs
+TARGETDIR=$APPSVC_ROOT/target
+
+# If the libs don't exist, or it's modification time is older than the last commit in $LIBSDIR/ios, wipe it out.
+if [ ! -d $LIBSDIR/ios ] || [ $(stat -f "%m" $LIBSDIR/ios) -lt $(git log -n 1 --pretty=format:%at -- $LIBSDIR) ]; then
```

Comment:
That git foo line is pretty smart, we should probably use it everywhere else (e.g. our build-all* scripts) instead of just testing for folder existence.

Reply:
Honestly between this and downloading artifacts, there isn't a whole lot of benefit. I guess local desktop builds. I'll file a follow up.